controlplane/pkg/nsmd/nsmd.go

```diff
@@ -280,6 +264,46 @@ func (nsm *nsmServer) restoreClients(registeredEndpoints *registry.NetworkServic
 	logrus.Infof("NSMD: Restore of NSE/Clients Complete...")
 }
 
+func (nsm *nsmServer) isEndpointAlive(workspace string, name string) (bool, error) {
+	ws, ok := nsm.workspaces[workspace]
+	if !ok {
+		return false, fmt.Errorf("workspace %v doesn't exist", workspace)
+	}
+
+	ctx, cancelCtx := context.WithTimeout(context.Background(), NSEAliveTimeout)
+	defer cancelCtx()
+
+	logrus.Infof("Checking NSE %s is alive at %v...", name, ws.NsmClientSocket())
+	if nseConn, err := tools.SocketOperationCheckContext(ctx, tools.SocketPath(ws.NsmClientSocket())); err != nil {
+		return false, nil
+	} else {
+		_ = nseConn.Close()
+	}
+
+	logrus.Infof("NSE %s is alive at %v...", name, ws.NsmClientSocket())
+
+	return true, nil
+}
+
+func (nsm *nsmServer) deleteEndpoint(name string) error {
+	client, err := nsm.serviceRegistry.NseRegistryClient()
+	if err != nil {
+		return err
+	}
+
+	return nsm.deleteEndpointWithClient(name, client)
+}
+
+func (nsm *nsmServer) deleteEndpointWithClient(name string, client registry.NetworkServiceRegistryClient) error {
```

Comment:
Maybe you know what happens with endpoint's workspace? Is it deleted by DevicePlugin, or we should delete workspace here?

Reply:
As discussed offline, workspace should be deleted by DevicePlugin.